pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -498,7 +568,7 @@ func GetPipelineConditionStatus(pr *v1beta1.PipelineRun, state PipelineRunState,
 
 	// Hasn't timed out; not all tasks have finished....
 	// Must keep running then....
-	if failedTasks > 0 || cancelledTasks > 0 {
+	if cancelledTasks > 0 || (failedTasks > 0 && state.checkTasksDone(dfinally)) {
```

Comment:
> @pritidesai : this additional check is making sure that pipeline does not exit when final tasks are not done executing.
> Without this additional check state.isFinalTasksDone(dfinally), GetPipelineConditionStatus changes the pipeline status to stopping irrespective of where the final tasks are in their execution cycle (scheduled, started, running, failed, succeeded, retrying etc)

Thanks for explaining - can you add a comment explaining this?

I think it might also be worth explaining this progression in the markdown docs.

Reply:
hey @bobcatfish I have updated [doc](https://github.com/tektoncd/pipeline/pull/2661/files#diff-db9ac78ce08367ea2b7db8c1412049b6R628) and added [comment](https://github.com/tektoncd/pipeline/pull/2661/files#diff-4e301064fef70f3c0db810267c0959f2R570) explaining this.